test/testdata/desugar/range.rb

```diff
@@ -14,9 +14,39 @@ def foo
     T.reveal_type(e) # error: Revealed type: `Integer`
     T.reveal_type(f) # error: Revealed type: `String`
 
-    # testing for endless ranges
-    g = (1..)
-    h = Range.new(1, nil)
+    # testing for beginless ranges
+    g = (..1)
+    h = Range.new(nil, 1)
     T.reveal_type(g) # error: Revealed type: `T::Range[Integer(1)]`
     T.reveal_type(h) # error: Revealed type: `T::Range[Integer(1)]`
+
+    # testing for endless ranges
+    i = (1..)
+    j = Range.new(1, nil)
+    T.reveal_type(i) # error: Revealed type: `T::Range[Integer(1)]`
+    T.reveal_type(j) # error: Revealed type: `T::Range[Integer(1)]`
```

Comment:
This doesn't seem quite right to me, The range will produce values other than `1`, right?

I'm not entirely sure if this is something that we can fix with an rbi change, so if you can't fix it it might be easier to look at adding an intrinsic in `core/types/calls.cc`.

Reply:
Oh you're right, I didn't realized it was using `1` everywhere.

Yeah I think at this point I reached the limit of the RBI power. I'll add a special case for ranges.